src/objects.js

```diff
@@ -5788,6 +5788,24 @@ SpriteMorph.prototype.allMessageNames = function () {
     return msgs;
 };
 
+SpriteMorph.prototype.allSendersOf = function (message) {
+    if (typeof message === 'number') {message = message.toString(); }
+    return this.scripts.children.filter(morph => {
+        var event;
+        if (morph.selector) {
+            if (
+                contains(['doBroadcast', 'doBroadcastAndWait'], morph.selector)
+            ) {
+                event = morph.inputs()[0].evaluate();
```

Comment:
Not sure if *evaluating* the first input always works, e.g. if it's an embedded reporter. In this case something might go awry. Probably best to check whether the first input is an InputSlotMorph, before evaluating it.

Reply:
Thanks, done too!